server/queues/processors/WebsocketsProcessor.ts

```diff
@@ -278,63 +282,73 @@ export default class WebsocketsProcessor {
-            event: event.name,
-            userId: groupUser.userId,
-            collectionId: event.collectionId,
-          });
-          // tell any user clients to connect to the websocket channel for the collection
-          socketio.to(`user-${groupUser.userId}`).emit("join", {
-            event: event.name,
-            collectionId: event.collectionId,
-          });
+        const membership = await GroupMembership.findByPk(
+          event.data.membershipId
+        );
+        if (!membership) {
+          return;
         }
 
+        socketio
+          .to(`group-${membership.groupId}`)
+          .emit(event.name, presentGroupMembership(membership));
+
+        socketio
+          .to(`collection-${membership.collectionId}`)
+          .emit(event.name, presentGroupMembership(membership));
+
+        socketio.to(`group-${membership.groupId}`).emit("join", {
+          event: event.name,
+          collectionId: event.collectionId,
+        });
+
         return;
       }
 
       case "collections.remove_group": {
-        const [groupUsers, membershipUserIds] = await Promise.all([
-          GroupUser.findAll({
-            where: { groupId: event.modelId },
-          }),
-          Collection.membershipUserIds(event.collectionId),
-        ]);
+        const membership = {
+          groupId: event.modelId,
+          collectionId: event.collectionId,
+          id: event.data.membershipId,
+        };
 
-        for (const groupUser of groupUsers) {
-          if (membershipUserIds.includes(groupUser.userId)) {
-            // the user still has access through some means...
-            // treat this like an add, so that the client re-syncs policies
-            socketio
-              .to(`user-${groupUser.userId}`)
-              .emit("collections.add_user", {
-                event: event.name,
-                userId: groupUser.userId,
-                collectionId: event.collectionId,
-              });
-          } else {
-            // let users in the channel know they were removed
-            socketio
-              .to(`user-${groupUser.userId}`)
-              .emit("collections.remove_user", {
-                event: event.name,
-                userId: groupUser.userId,
-                collectionId: event.collectionId,
-              });
-            // tell any user clients to disconnect to the websocket channel for the collection
-            socketio.to(`user-${groupUser.userId}`).emit("leave", {
-              event: event.name,
-              collectionId: event.collectionId,
-            });
+        // let everyone with access to the collection know a group was removed
+        socketio
+          .to(`collection-${event.collectionId}`)
+          .emit("collections.remove_group", membership);
+
+        // let everyone in the group know they were removed
+        socketio
+          .to(`group-${event.modelId}`)
+          .emit("collections.remove_group", membership);
```

Comment:
Everyone in `group-${event.modelId}` would still be members of the `collection-${event.collectionId}` room at this point. So this is perhaps redundant?